### ISSUE TITLE
screen shaking vigorously while scrolling up or down due to pull hook to refresh

### ISSUE THREAD

**Issue by eljavidelomas** [OnsenUI/OnsenUI#917]
The problem is that when I hold the screen and scroll up or down, the screen starts shaking vigorously and verified that when I remove the pull hook the scroll was fine, but when I insert the pull hook again, the shaking appear again.
I'm testing my app on Android 4.1.2, compiled with phonegap. I have the next code:

``` html
<ons-carousel-item id="ci-Calendario">
        <ons-page >
                <ons-pull-hook ng-action="actualizar($done)" var="loader" height="80px" threshold-height="-220px" >
                        <span ng-switch="loader.getCurrentState()">
                                <span ng-switch-when="initial"><ons-icon size="35px" icon="ion-arrow-down-a"></ons-icon> Deslice hacia abajo para Actualizar</span>
                                <span ng-switch-when="preaction"><ons-icon size="35px" icon="ion-arrow-up-a"></ons-icon> Suelte para Actualizar</span>
                                <span class="mensajePullHookContenedor" ng-switch-when="action"><ons-icon size="35px" spin="true" icon="ion-load-d"></ons-icon> Actualizando...</span>

                        </span>
                   </ons-pull-hook>
                   <div id="boxscroll" > <!-- Here is the content that i scroll up and down-->
                            <div id='calendar'></div>
                            <div class="col-md-10 col-vlg-7">
                                <ul class="cbp_tmtimeline" style="font-size:13px"></ul>
                            </div>
                     </div>
           </ons-page>           
  </ons-carousel-item>
```

I reviewed the code of the pull hook and found that in the method "_on_drag", line 11173 of the file onsenui.js of version 1.3.10 are certain code that cause the problem.
Could you check that issue ??

This is the code of the file "onseui.js"  that i think is making the shaking:

``` javascript
_onDrag: function(event) {
        if (this.isDisabled()) {
          return;
        }

        // Ignore when dragging left and right.
        if (event.gesture.direction === 'left' || event.gesture.direction === 'right') {
          return;
        }

        // Hack to make it work on Android 4.4 WebView. Scrolls manually near the top of the page so
        // there will be no inertial scroll when scrolling down. Allowing default scrolling will
        // kill all 'touchmove' events.
        var el = this._pageElement[0];
        el.scrollTop = this._startScroll - event.gesture.deltaY;
        if (el.scrollTop < window.innerHeight && event.gesture.direction !== 'up') {
          event.gesture.preventDefault();
        }

        if (this._currentTranslation === 0 && this._getCurrentScroll() === 0) {
          this._transitionDragLength = event.gesture.deltaY;

          var direction = event.gesture.interimDirection;
          if (direction === 'down') {
            this._transitionDragLength -= 1;// <-- This line maybe are provoking the shaking 
          }
          else {
            this._transitionDragLength += 1;//<-- This line maybe are provoking the shaking 
          }
        }

        var scroll = event.gesture.deltaY - this._startScroll;

        scroll = Math.max(scroll, 0);

        if (this._thresholdHeightEnabled() && scroll >= this._getThresholdHeight()) {
          event.gesture.stopDetect();

          setImmediate(function() {
            this._setState(this.STATE_ACTION);
            this._translateTo(this._getHeight(), {animate: true});

            this._waitForAction(this._onDone.bind(this));
          }.bind(this));
        }
        else if (scroll >= this._getHeight()) {
          this._setState(this.STATE_PREACTION);
        }
        else {
          this._setState(this.STATE_INITIAL);
        }

        event.stopPropagation();
        this._translateTo(scroll);
      },
```

Could you check that issue please??

Thanks !

**Comment by argelius** [comment]
Hi @eljavidelomas !

Can you make a small example that reproduces this?

How are you using the carousel? Is it a full screen carousel?

**Comment by mrboo00** [comment]
Any solution to this?

**Comment by Hirbod** [comment]
@argelius  I can confirm this and it's related to some of my reported problems on ons-pull-hook. The vigorously shaking is only reproducible on an device not inside the simulator.

Just make a big list and drag slowly, do not scroll fast. It will shake up and down.
Also, scroll fast, and try to give your inertal some more speed, as soon as you touch the screen, it will jump to the last starting positing and will return back immediately.

Performance is also very poor when the list is big. It's not usable actually. I've integrated something by myself now (based on -px scroll values) - will only work on iOS but won't require transform or any drag events.. This should be improved.

**Comment by argelius** [comment]
Thanks @Hirbod !

I will investigate the behavior on iOS.

**Comment by ychongsaytc** [comment]
Why was this closed? Still happens in version 2.9.2.

**Comment by asialgearoid** [comment]
@ychongsaytc If this is still happening, please open a new issue. It is hard to keep track of comments on very old issues.
